core/src/main/java/org/infinispan/persistence/spi/InitializationContext.java

```diff
@@ -51,17 +51,18 @@
    ExecutorService getExecutor();
 
    /**
-    * Returns an executor that Infinispan uses internally for non blocking tasks. The user must guarantee tasks
-    * submitted to this executor will not block the thread it is ran on. Failure to do so can slow down Infinispan's
-    * handling of operations as these threads are limited to the number of cores and are used extensively.
-    * @return an executor that can be used to submit tasks that will not block the thread it runs on
+    * Returns an executor that Infinispan uses to perform tasks in a non-blocking manner. Users must guarantee that the tasks they
+    * submit to this executor do not block the thread in which the executor runs. Doing so can cause Infinispan to handle operations
+    * more slowly, reducing performance, because threads are limited to the number of cores and are used extensively.
+    * @return an executor that can submit non-blocking tasks.
     */
    Executor getNonBlockingExecutor();
 
    /**
-    * Returns a manager that is designed to execute tasks that may block. This manager ensures that only the blocking
-    * portion is ran on a blocking thread and any stage is continued on a non blocking thread.
-    * @return a manager that should be used to execute blocking operations
+    * Returns a manager that can execute thread-blocking operations. The manager runs any blocking operations on blocking threads,
+    * any stages continue on non-blocking threads. Use this manager to ensure that blocking operations do not leak into the
+    * Infinispan internal system.
+    * @return a manager that can execute blocking operations.
```

Comment:
IMO the original better conveys the purpose of the `BlockingManager`.

Reply:
thanks @ryanemerson I've changed back to the original (with a couple minor tweaks)